File: ios/VKAdsModule.swift
import ExpoModulesCore

/**
 * VK Ads Module — iOS stub.
 *
 * The myTarget SDK is currently only available for Android.
 * This iOS implementation is a no-op placeholder.
 *
 * For iOS VK Ads integration, use the myTarget iOS SDK directly:
 * https://github.com/myTargetSDK/mytarget-ios
 */
public class VKAdsModule: Module {
    public func definition() -> ModuleDefinition {
        Name("VKAdsModule")

        // --- Initialization (no-op on iOS) ---
        Function("initialize") {
            print("[VKAds] iOS: SDK not available. Use mytarget-ios directly.")
        }

        Function("setDebugMode") { (enabled: Bool) in
            print("[VKAds] iOS: debug mode set to \(enabled)")
        }

        // --- Banner (no-op on iOS) ---
        Function("loadBanner") { (slotId: Int, adSize: String, refreshAd: Bool) -> String in
            print("[VKAds] iOS: Banner not supported. Use mytarget-ios directly.")
            return ""
        }

        Function("reloadBanner") { (identifier: String) in
            print("[VKAds] iOS: Banner reload not supported.")
        }

        Function("setBannerCustomParams") { (identifier: String, age: Int, gender: Int, email: String, phone: String) in
            print("[VKAds] iOS: Custom params not supported.")
        }

        // --- Interstitial (no-op on iOS) ---
        AsyncFunction("loadInterstitial") { (slotId: Int, promise: Promise) in
            print("[VKAds] iOS: Interstitial not supported.")
            promise.resolve("")
        }

        AsyncFunction("showInterstitial") { (identifier: String, promise: Promise) in
            print("[VKAds] iOS: Interstitial show not supported.")
            promise.resolve(nil)
        }

        Function("setInterstitialCustomParams") { (identifier: String, age: Int, gender: Int, email: String, phone: String) in
            print("[VKAds] iOS: Custom params not supported.")
        }

        // --- Rewarded (no-op on iOS) ---
        AsyncFunction("loadRewarded") { (slotId: Int, promise: Promise) in
            print("[VKAds] iOS: Rewarded not supported.")
            promise.resolve("")
        }

        AsyncFunction("showRewarded") { (identifier: String, promise: Promise) in
            print("[VKAds] iOS: Rewarded show not supported.")
            promise.resolve(nil)
        }

        Function("setRewardedCustomParams") { (identifier: String, age: Int, gender: Int, email: String, phone: String) in
            print("[VKAds] iOS: Custom params not supported.")
        }
    }
}

File: src/VKAdsModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import type { VKAdsBannerEvents, VKAdsCustomParams, VKAdSize } from './VKAds.types';

declare class VKAdsModule extends NativeModule {
  /**
   * Initialize the VK Ads SDK. Call this once at app startup.
   */
  initialize(): void;

  /**
   * Set debug mode for testing. Disable in production.
   */
  setDebugMode(enabled: boolean): void;

  // --- Banner ---
  /**
   * Load a banner ad for the given slot ID and size.
   * Returns a unique identifier for the banner.
   */
  loadBanner(slotId: number, adSize: string, refreshAd: boolean): string;

  /**
   * Reload an existing banner.
   */
  reloadBanner(identifier: string): void;

  /**
   * Set custom targeting params for a banner.
   */
  setBannerCustomParams(
    identifier: string,
    age: number,
    gender: number,
    email: string,
    phone: string
  ): void;

  // --- Interstitial ---
  /**
   * Create and load an interstitial ad.
   */
  loadInterstitial(slotId: number): Promise<string>;

  /**
   * Show a loaded interstitial ad.
   */
  showInterstitial(identifier: string): Promise<void>;

  /**
   * Set custom targeting params for an interstitial.
   */
  setInterstitialCustomParams(
    identifier: string,
    age: number,
    gender: number,
    email: string,
    phone: string
  ): void;

  // --- Rewarded ---
  /**
   * Create and load a rewarded video ad.
   */
  loadRewarded(slotId: number): Promise<string>;

  /**
   * Show a loaded rewarded ad.
   */
  showRewarded(identifier: string): Promise<void>;

  /**
   * Set custom targeting params for a rewarded ad.
   */
  setRewardedCustomParams(
    identifier: string,
    age: number,
    gender: number,
    email: string,
    phone: string
  ): void;

  // --- Events ---
  /** Events for banner ads (identifier-based) */
  bannerEvents: Record<string, VKAdsBannerEvents>;
  /** Events for interstitial ads */
  interstitialEvents: Record<string, VKAdsBannerEvents>;
  /** Events for rewarded ads */
  rewardedEvents: Record<string, VKAdsBannerEvents>;
}

export default requireNativeModule<VKAdsModule>('VKAdsModule');
